packages/widget_toolkit/lib/src/base/theme/widget_toolkit_theme.dart

```diff
@@ -6,585 +6,717 @@ import 'design_system/widget_toolkit_design_system.dart';
-  ];
-
-  static List<Color> searchTextFieldIconColorActive = [
-    WidgetToolkitDesignSystem.light().colors.blue,
-    WidgetToolkitDesignSystem.dark().colors.darkBlue,
-  ];
-
-  static List<Color> searchTextFieldBackgroundColor = [
-    WidgetToolkitDesignSystem.light().colors.white,
-    WidgetToolkitDesignSystem.dark().colors.darkGray,
-  ];
-
-  static List<Color> searchTextFieldBackgroundColorActive = [
-    WidgetToolkitDesignSystem.light().colors.mediumWhite,
-    WidgetToolkitDesignSystem.dark().colors.darkGray,
-  ];
-
-  static List<double> searchTextFieldBorderRadius = [
-    WidgetToolkitDesignSystem.light().spacings.xs,
-    WidgetToolkitDesignSystem.light().spacings.xs,
-  ];
-
-  static List<BoxBorder> searchTextFieldBorderType = [
-    Border.all(
-      color: WidgetToolkitDesignSystem.light().colors.mediumWhite,
-      width: WidgetToolkitDesignSystem.light().spacings.xxxs,
-    ),
-    Border.all(
-      color: WidgetToolkitDesignSystem.light().colors.mediumWhite,
-      width: WidgetToolkitDesignSystem.light().spacings.xxxs,
-    ),
-  ];
-
-  static List<EdgeInsets> searchTextFieldIconEdgeInsets = [
-    EdgeInsets.symmetric(
-      horizontal: WidgetToolkitDesignSystem.light().spacings.m,
-      vertical: WidgetToolkitDesignSystem.light().spacings.s,
-    ),
-    EdgeInsets.symmetric(
-      horizontal: WidgetToolkitDesignSystem.light().spacings.m,
-      vertical: WidgetToolkitDesignSystem.light().spacings.s,
-    ),
-  ];
-
-  static List<TextStyle> searchTextFieldHintStyle = [
-    WidgetToolkitDesignSystem.light().typography.h3Reg14.copyWith(
-          color: WidgetToolkitDesignSystem.dark().colors.gray,
+@TailorMixinComponent()
+class WidgetToolkitTheme extends ThemeExtension<WidgetToolkitTheme>
+    with _$WidgetToolkitThemeTailorMixin {
+  const WidgetToolkitTheme({
+    required this.primaryColor,
+    required this.backgroundColor,
+    required this.scaffoldBackgroundColor,
+    required this.highlightColor,
+    required this.primaryGradientStart,
+    required this.primaryGradientEnd,
+    required this.searchTextFieldIconColor,
+    required this.searchTextFieldIconColorActive,
+    required this.searchTextFieldBackgroundColor,
+    required this.searchTextFieldBackgroundColorActive,
+    required this.searchTextFieldBorderRadius,
+    required this.searchTextFieldBorderType,
+    required this.searchTextFieldIconEdgeInsets,
+    required this.searchTextFieldHintStyle,
+    required this.searchTextFieldTextStyle,
+    required this.pickerListItemInnerEdgeInsets,
+    required this.pickerListItemOuterEdgeInsets,
+    required this.pickerListItemBorderRadius,
+    required this.pickerListItemSelectedColor,
+    required this.pickerListItemUnselectedColor,
+    required this.pickerListItemTextStyle,
+    required this.captionBold,
+    required this.textButtonTextStyle,
+    required this.errorCardIconColor,
+    required this.errorCardBackgroundColor,
+    required this.lightRed,
```

Comment:
Please rename this property to its domain ... it will be confusing the set a `blue` color to a property, which is called `lightRed`

Reply:
This applies for all non-domain specifics properties